### PR TITLE
Add Apalache server integration

### LINKED ISSUE #822: Generate gRPC glue code for Apalache server
This will look approximately like https://github.com/informalsystems/apalache-chai/pull/6

### PR DESCRIPTION
**Pull request by shonfeder**
Closes #822 

This changest is focused on providing the minimal functionality needed for integration with the apalache gRPC server.

Unfortunately, due the the opsy nature of this goal, I've had to touch quite a few places. Sorry for the big diff.

The good news is: 3/4th of the diff is either package updates or relatively low density integration tests. So it's really not as bad as it looks :)

Much functionality is yet to be added: but this foundation will let us add the next steps in small iterations (such as more useful error handing, transparent management of apalache, and production of ITF traces).

- [x] Tests added for any new code
- [x] Ran `npm run format` (or had formatting run automatically on all files edited)
- [x] Documentation added for any new functionality
- [x] Entries added to the respective `CHANGELOG.md` for any new functionality
- [x] Feature table on [`README.md`](../README.md#roadmap) updated for any listed functionality

### CODE REVIEW

File: quint/apalache-tests.md [hunk 1]
```diff
@@ -1,74 +1,75 @@
...
 ```
 
-## Can check `../examples/language-features/integers.qnt`
 
-<!-- !test in can check integers.qnt -->
-```
-quint typecheck --out _build/integers.qnt.json ../examples/language-features/integers.qnt
-apalache-mc check _build/integers.qnt.json | grep -o "EXITCODE: OK"
-```
+## Successful verification
```
Comment (shonfeder) [inline quint/apalache-tests.md:36]: The following tests are just mechanical replacement of the manual cli invocations with invocation of the `verify` command.

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -98,10 +98,19 @@ jobs:
           distribution: "temurin" # See 'Supported distributions' for available options
           java-version: "19"
       - name: Apalache integration tests
-        run: cd ./quint && npm run apalache-integration
+        run: |
+          cd ./quint
+          # Fetch the latest apalache release
+          make -C .. apalache
+          # Run the "disintegration" tests
```
Comment (konnov) [inline .github/workflows/main.yml:105]:
```suggestion
          # Run the "integration" tests
```

Comment (konnov) [reply .github/workflows/main.yml:105]: What are "disintegration" tests? :)

File: quint/README.md [hunk 3]
```diff
@@ -5,7 +5,7 @@ with [the Quint specification language](https://github.com/informalsystems/quint
 
 ## Dependencies
 
-- Node.js >= 14
+- Node.js >= 18
```
Comment (konnov) [inline quint/README.md:8]: Is it a common version of nodejs?

File: .github/workflows/main.yml (hunk 2 above)
Comment (shonfeder) [reply .github/workflows/main.yml:105]: They test ways that the integration fails :)

File: quint/README.md (hunk 3 above)
Comment (shonfeder) [reply quint/README.md:8]: Yes! This is the currently active LTS version. This addresses #867. See https://nodejs.dev/en/about/releases/

14 is already beyond its maintenance window.

**Comment by shonfeder** [comment]
Thanks for the review1

**Comment by shonfeder** [comment]
I thought this was merged already!

**Pull request merged**